lib/hooks/useTheme.tsx

```diff
@@ -1,23 +1,22 @@
-import { useEffect, useState } from "react";
+import { useLayoutEffect, useState } from "react";
 
 import { Maybe } from "@trpc/server";
 
 // makes sure the ui doesn't flash
 export default function useTheme(theme?: Maybe<string>) {
   const [isReady, setIsReady] = useState(false);
-  useEffect(() => {
-    setIsReady(true);
-    if (!theme) {
```

Comment:
The problem was here. `theme` is `null` when the option is checked so the logic was ignored.
It can be removed because the same check is done on line 15 (or line 12 after the changes).